<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>885</width>
    <height>666</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Recon3dX</string>
  </property>
  <property name="windowIcon">
   <iconset resource="res.qrc">
    <normaloff>:/logo.png</normaloff>:/logo.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <item>
     <widget class="QWidget" name="widget_4" native="true">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>320</height>
       </size>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_7">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QWidget" name="widget" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="styleSheet">
          <string notr="true"/>
         </property>
         <layout class="QVBoxLayout">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QLabel" name="label">
              <property name="text">
               <string>辅助相机</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="cb_cam_mark"/>
            </item>
            <item>
             <widget class="QLabel" name="lb_cam_mark">
              <property name="text">
               <string>[0x0]</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="ImageLabel" name="lb_img_mark">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="styleSheet">
             <string notr="true">background-color: rgb(0, 0, 0);</string>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="widget_2" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="styleSheet">
          <string notr="true"/>
         </property>
         <layout class="QVBoxLayout" name="_2">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>主相机(左)</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="cb_cam_left"/>
            </item>
            <item>
             <widget class="QLabel" name="lb_cam_left">
              <property name="text">
               <string>[0x0]</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="ImageLabel" name="lb_img_left">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="styleSheet">
             <string notr="true">background-color: rgb(0, 0, 0);</string>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="widget_3" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="styleSheet">
          <string notr="true"/>
         </property>
         <layout class="QVBoxLayout" name="_3">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_5">
            <item>
             <widget class="QLabel" name="label_7">
              <property name="text">
               <string>主相机(右)</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="cb_cam_right"/>
            </item>
            <item>
             <widget class="QLabel" name="lb_cam_right">
              <property name="text">
               <string>[0x0]</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="ImageLabel" name="lb_img_right">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="styleSheet">
             <string notr="true">background-color: rgb(0, 0, 0);</string>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_13">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>投影仪</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_9">
           <item>
            <layout class="QGridLayout" name="gridLayout">
             <item row="0" column="0">
              <widget class="QLabel" name="label_11">
               <property name="minimumSize">
                <size>
                 <width>100</width>
                 <height>30</height>
                </size>
               </property>
               <property name="text">
                <string>条纹宽度(pixel)</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="label_12">
               <property name="minimumSize">
                <size>
                 <width>100</width>
                 <height>30</height>
                </size>
               </property>
               <property name="text">
                <string>亮度</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QLineEdit" name="le_strip_brightness">
               <property name="minimumSize">
                <size>
                 <width>0</width>
                 <height>30</height>
                </size>
               </property>
               <property name="text">
                <string>255</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLineEdit" name="le_strip_w">
               <property name="minimumSize">
                <size>
                 <width>0</width>
                 <height>30</height>
                </size>
               </property>
               <property name="text">
                <string>5</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QPushButton" name="btn_project">
             <property name="minimumSize">
              <size>
               <width>100</width>
               <height>66</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>100</width>
               <height>66</height>
              </size>
             </property>
             <property name="text">
              <string>显示结构光</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_3">
          <property name="title">
           <string>手持设备</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_11">
           <item>
            <layout class="QGridLayout" name="gridLayout_2">
             <property name="verticalSpacing">
              <number>0</number>
             </property>
             <item row="0" column="1">
              <widget class="QLineEdit" name="le_mark_w">
               <property name="minimumSize">
                <size>
                 <width>0</width>
                 <height>30</height>
                </size>
               </property>
               <property name="text">
                <string>100</string>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QLabel" name="label_14">
               <property name="minimumSize">
                <size>
                 <width>100</width>
                 <height>30</height>
                </size>
               </property>
               <property name="text">
                <string>立方体边长(mm)</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QLineEdit" name="le_mark_ww">
               <property name="minimumSize">
                <size>
                 <width>0</width>
                 <height>30</height>
                </size>
               </property>
               <property name="text">
                <string>28</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="label_15">
               <property name="minimumSize">
                <size>
                 <width>100</width>
                 <height>30</height>
                </size>
               </property>
               <property name="text">
                <string>棋盘格边长(mm)</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QPushButton" name="btn_mark">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>100</width>
               <height>66</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>100</width>
               <height>66</height>
              </size>
             </property>
             <property name="text">
              <string>标定位置</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_6">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>60</height>
           </size>
          </property>
          <property name="title">
           <string>相机</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_14">
           <item>
            <widget class="QLabel" name="label_19">
             <property name="text">
              <string>参数文件</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="le_param_path">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>30</height>
              </size>
             </property>
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btn_choose_param">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>30</height>
              </size>
             </property>
             <property name="text">
              <string>选择</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="styleSheet">
         <string notr="true"/>
        </property>
        <property name="title">
         <string>重建</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="0" column="0">
          <widget class="QPushButton" name="btn_save_img">
           <property name="minimumSize">
            <size>
             <width>150</width>
             <height>50</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>150</width>
             <height>50</height>
            </size>
           </property>
           <property name="text">
            <string>保存图像</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QPushButton" name="btn_recon">
           <property name="minimumSize">
            <size>
             <width>150</width>
             <height>50</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>150</width>
             <height>50</height>
            </size>
           </property>
           <property name="text">
            <string>开始重建</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QPushButton" name="btn_show_left">
           <property name="minimumSize">
            <size>
             <width>150</width>
             <height>50</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>150</width>
             <height>50</height>
            </size>
           </property>
           <property name="text">
            <string>左图像</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QPushButton" name="btn_show_right">
           <property name="minimumSize">
            <size>
             <width>150</width>
             <height>50</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>150</width>
             <height>50</height>
            </size>
           </property>
           <property name="text">
            <string>右图像</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QPushButton" name="btn_show_rectify">
           <property name="minimumSize">
            <size>
             <width>150</width>
             <height>50</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>150</width>
             <height>50</height>
            </size>
           </property>
           <property name="text">
            <string>校正图</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QPushButton" name="btn_show_disparity">
           <property name="minimumSize">
            <size>
             <width>150</width>
             <height>50</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>150</width>
             <height>50</height>
            </size>
           </property>
           <property name="text">
            <string>视差图</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QPushButton" name="btn_save_res">
           <property name="minimumSize">
            <size>
             <width>150</width>
             <height>50</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>150</width>
             <height>50</height>
            </size>
           </property>
           <property name="text">
            <string>保存结果</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QPushButton" name="btn_show_pcl">
           <property name="minimumSize">
            <size>
             <width>150</width>
             <height>50</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>150</width>
             <height>50</height>
            </size>
           </property>
           <property name="text">
            <string>查看点云</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_4">
        <property name="title">
         <string>拼接</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <widget class="QPushButton" name="btn_add_pcl">
           <property name="minimumSize">
            <size>
             <width>150</width>
             <height>50</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>150</width>
             <height>50</height>
            </size>
           </property>
           <property name="text">
            <string>添加点云</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="lb_pcl_cnt">
           <property name="minimumSize">
            <size>
             <width>150</width>
             <height>50</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>150</width>
             <height>50</height>
            </size>
           </property>
           <property name="text">
            <string>点云个数：0</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btn_join">
           <property name="minimumSize">
            <size>
             <width>150</width>
             <height>50</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>150</width>
             <height>50</height>
            </size>
           </property>
           <property name="text">
            <string>开始拼接</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btn_grid">
           <property name="minimumSize">
            <size>
             <width>150</width>
             <height>50</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>150</width>
             <height>50</height>
            </size>
           </property>
           <property name="text">
            <string>三角网格构建</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>ImageLabel</class>
   <extends>QLabel</extends>
   <header>ImageLabel.hpp</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="res.qrc"/>
 </resources>
 <connections/>
</ui>
